Source/controls/plrctrls.cpp

```diff
@@ -1409,37 +1408,75 @@ bool IsStickMovementSignificant()
 	    || rightStickX != 0 || rightStickY != 0;
 }
 
-ControlTypes GetInputTypeFromEvent(const SDL_Event &event)
+ControlTypeWithGamepad GetInputTypeFromEvent(const SDL_Event &event)

```

Comment:
Adding Gamepad layout to this function both feels a bit wrong since not every control type is a gamepad, but also it makes the function a lot more complicated (a lot of it being repetitive as well).

Maybe instead it should just be it own function that gets called separately.